causal_testing/json_front/json_class.py

```diff
@@ -89,54 +88,58 @@ def setup(self):
+            if "effect_modifiers" in test
+            else {},
+            estimate_type=test["estimate_type"],
+        )
+        return abstract_test
+
+    def generate_tests(self, effects: dict, mutates: dict, estimators: dict, f_flag: bool):
         """Runs and evaluates each test case specified in the JSON input
 
         :param effects: Dictionary mapping effect class instances to string representations.
         :param mutates: Dictionary mapping mutation functions to string representations.
         :param estimators: Dictionary mapping estimator classes to string representations.
         :param f_flag: Failure flag that if True the script will stop executing when a test fails.
         """
-        executed_tests = 0
         failures = 0
         for test in self.test_plan["tests"]:
             if "skip" in test and test["skip"]:
                 continue
-
-            abstract_test = AbstractCausalTestCase(
-                scenario=self.modelling_scenario,
-                intervention_constraints=[mutates[v](k) for k, v in test["mutations"].items()],
-                treatment_variables={self.modelling_scenario.variables[v] for v in test["mutations"]},
-                expected_causal_effect={
-                    self.modelling_scenario.variables[variable]: effects[effect]
-                    for variable, effect in test["expectedEffect"].items()
-                },
-                effect_modifiers={self.modelling_scenario.variables[v] for v in test["effect_modifiers"]}
-                if "effect_modifiers" in test
-                else {},
-                estimate_type=test["estimate_type"],
-            )
+            abstract_test = self._create_abstract_test_case(test, mutates, effects)
 
             concrete_tests, dummy = abstract_test.generate_concrete_tests(5, 0.05)
             logger.info("Executing test: %s", test["name"])
             logger.info(abstract_test)
             logger.info([(v.name, v.distribution) for v in abstract_test.treatment_variables])
             logger.info("Number of concrete tests for test case: %s", str(len(concrete_tests)))
-            for concrete_test in concrete_tests:
-                executed_tests += 1
-                failed = self._execute_test_case(concrete_test, estimators[test["estimator"]], f_flag)
-                if failed:
-                    failures += 1
+            failures = self._execute_tests(concrete_tests, estimators, test, f_flag)
+
+        logger.info("{%d}/{%d} failed", failures, str(len(concrete_tests)))
 
-        logger.info("{%d}/{%d} failed", failures, executed_tests)
+    def _execute_tests(self, concrete_tests, estimators, test, f_flag):
+        failures = 0
+        for concrete_test in concrete_tests:
+            failed = self._execute_test_case(concrete_test, estimators[test["estimator"]], f_flag)
+            if failed:
+                failures += 1
+        return failures
 
     def _json_parse(self):
```

Comment:
After reading your comments on testing a mix of both public/private methods and how hard to follow that got. I've moved to just testing the public methods as 'units'. I couldn't find any other private method testing in the project so I think it will be nice to follow the rest of the projects testing as well.

Reply:
So this specific code is now covered in the testing of `generate_tests`